### PR TITLE
[ccma] fix extractor. A list of urls is now being sent from the server.

### PR DESCRIPTION
**Pull request by andrescarrasco**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence: it is part of youtube-dl)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

### CODE REVIEW

File: youtube_dl/extractor/ccma.py [hunk 1]
```diff
@@ -52,12 +52,22 @@ def _real_extract(self, url):
...
                 media_data = md
                 media_url = media_data.get('media', {}).get('url')
-                if media_url:
+                if media_url and isinstance(media_url, list):
+                    q = 0
+                    for _url in media_url:
+                        formats.append({
+                            'format_id': profile,
+                            'url': _url['file'],
+                            'quality': q,
```
Comment (dstftw) [inline youtube_dl/extractor/ccma.py]: No, parse height from label.

File: youtube_dl/extractor/ccma.py [hunk 2]
```diff
@@ -52,12 +52,22 @@ def _real_extract(self, url):
             if md:
                 media_data = md
                 media_url = media_data.get('media', {}).get('url')
-                if media_url:
+                if media_url and isinstance(media_url, list):
+                    q = 0
+                    for _url in media_url:
+                        formats.append({
+                            'format_id': profile,
+                            'url': _url['file'],
```
Comment (dstftw) [inline youtube_dl/extractor/ccma.py]: Must not break extraction if missing.

File: youtube_dl/extractor/ccma.py [hunk 3]
```diff
@@ -52,12 +52,22 @@ def _real_extract(self, url):
...
+                            'quality': q,
+                        })
+                        q += 1
+                elif media_url and isinstance(media_url, str):
```
Comment (dstftw) [inline youtube_dl/extractor/ccma.py]: `compat_str`.

File: youtube_dl/extractor/ccma.py (hunk 1 above)
Comment (andrescarrasco) [reply youtube_dl/extractor/ccma.py]: I'm sorry, but i don't understand what do you mean by height and label, could you elaborate?

Comment (dstftw) [reply youtube_dl/extractor/ccma.py]: `_url['label']`.

**Comment by andrescarrasco** [comment]
The changes should now address the changes requested.

File: youtube_dl/extractor/ccma.py [hunk 4]
```diff
@@ -52,12 +53,30 @@ def _real_extract(self, url):
...
                 media_url = media_data.get('media', {}).get('url')
-                if media_url:
+                ext = media_data.get('media', {}).get('format')
+                if ext:
+                    ext = ext.lower()
+
+                if media_url and isinstance(media_url, list):
+                    for _url in media_url:
+                        if 'label' in _url:
+                            height = int_or_none(_url.get('label').replace('p', ''))
```
Comment (dstftw) [inline youtube_dl/extractor/ccma.py]: `utils.parse_resolution`.

File: youtube_dl/extractor/ccma.py [hunk 5]
```diff
@@ -52,12 +53,30 @@ def _real_extract(self, url):
             if md:
                 media_data = md
                 media_url = media_data.get('media', {}).get('url')
-                if media_url:
+                ext = media_data.get('media', {}).get('format')
+                if ext:
+                    ext = ext.lower()
```
Comment (dstftw) [inline youtube_dl/extractor/ccma.py]: It's already contained in video URL. Remove.

File: youtube_dl/extractor/ccma.py [hunk 6]
```diff
@@ -73,8 +92,8 @@ def _real_extract(self, url):
             if sub_url:
                 subtitles.setdefault(
                     subtitols.get('iso') or subtitols.get('text') or 'ca', []).append({
-                        'url': sub_url,
-                    })
+                    'url': sub_url,
+                })
```
Comment (dstftw) [inline youtube_dl/extractor/ccma.py]: Don't touch this.

File: youtube_dl/extractor/ccma.py [hunk 7]
```diff
@@ -52,12 +53,30 @@ def _real_extract(self, url):
...
+                if media_url and isinstance(media_url, list):
+                    for _url in media_url:
+                        if 'label' in _url:
+                            height = int_or_none(_url.get('label').replace('p', ''))
+                        else:
+                            height = None
+
+                        formats.append({
+                            'format_id': profile,
+                            'url': _url.get('file'),
```
Comment (dstftw) [inline youtube_dl/extractor/ccma.py]: URL must not be `None`.

Comment (andrescarrasco) [reply youtube_dl/extractor/ccma.py]: Do I default to an empty string then?

Comment (dstftw) [reply youtube_dl/extractor/ccma.py]: The latter.

**Comment by andrescarrasco** [comment]
@dstftw I've made the changes you requested, please take a look

**Pull request closed without merging**